maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/NewMavenJSFModuleProjectProvider.java

```diff
@@ -0,0 +1,278 @@
+
+        archetype.setGroupId( gav[0] );
+        archetype.setArtifactId( gav[1] );
+        archetype.setVersion( archetypeVersion );
+        final Properties properties = new Properties();
+
+        final ResolverConfiguration resolverConfig = new ResolverConfiguration();
+        ProjectImportConfiguration configuration = new ProjectImportConfiguration( resolverConfig );
+
+        final List<IProject> newProjects = projectConfigurationManager.createArchetypeProjects(
+            location, archetype, groupId, artifactId, version, "com.example", properties, configuration, monitor );
+
+        if( newProjects == null || newProjects.size() == 0 )
+        {
+            retval = LiferayMavenCore.createErrorStatus( "Unable to create project from archetype." );
+        }
+        else
+        {
+            for( IProject newProject : newProjects )
+            {
+                IFile gradleFile = newProject.getFile( "build.gradle" );
+
+                if( gradleFile.exists() )
+                {
+                    gradleFile.delete( true, monitor );
+                }
+            }
+
+            retval = Status.OK_STATUS;
+        }
+
+        return retval;
+    }
+
+    private Version extractLatestVersion( final String artifactId )
+    {
+        Version maxVersion = null;
+        try
+        {
+            String responseString = getHttpResponse(
+                "http://search.maven.org/solrsearch/select?q=g:com.liferay.faces.archetype+AND+a:" + artifactId +
+                    "&rows=20&wt=json" );
+            Object result = getJSONResponse( responseString );
+
+            if( result instanceof JSONObject )
+            {
+                JSONObject resultJson = (JSONObject) result;
+                Object response = resultJson.get( "response" );
+
+                if( response != null )
+                {
+                    JSONObject responseJson = (JSONObject) response;
+                    Object docs = responseJson.get( "docs" );
+
+                    if( docs != null && docs instanceof JSONArray )
+                    {
+                        JSONArray docsJson = (JSONArray) docs;
+
+                        for( int i = 0; i < docsJson.length(); i++ )
+                        {
+                            JSONObject docJson = (JSONObject) docsJson.get( i );
+                            String versionString = (String) docJson.get( "latestVersion" );
+                            Version tmpVersion = Version.parseVersion( versionString );
+
+                            if( tmpVersion.compareTo( maxVersion ) > 0 || maxVersion == null )
+                            {
+                                maxVersion = tmpVersion;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            LiferayMavenCore.logError( "Failed to get latest JSF archtype version", e );
```

Comment:
We need to have a fallback version that we can use if we fail to get the ones from online.